src/IFrameNavigator.ts

```diff
@@ -1,158 +1,388 @@
+
+    private handlePreviousPageClick(event: MouseEvent): void {
+        event.preventDefault();
+        if (this.paginator && !this.checkForIFrameLink(event)) {
+            if (this.paginator.onFirstPage()) {
+                if (this.previousChapterLink.hasAttribute("href")) {
+                    let position = {
+                        resource: this.previousChapterLink.href,
+                        position: 1
+                    };
+                    this.navigate(position);
+                }
+            } else {
+                this.paginator.goToPreviousPage();
+                this.saveCurrentReadingPosition();
+            }
+        }
+    }
+
+    private handleNextPageClick(event: MouseEvent) {
+        event.preventDefault();
+        if (this.paginator && !this.checkForIFrameLink(event)) {
+            if (this.paginator.onLastPage()) {
+                if (this.nextChapterLink.hasAttribute("href")) {
+                    let position = {
+                        resource: this.nextChapterLink.href,
+                        position: 0
+                    };
+                    this.navigate(position);
+                }
+            } else {
+                this.paginator.goToNextPage();
+                this.saveCurrentReadingPosition();
+            }
+        }
+    }
+
+    private handleResize(): void {
+        if (this.paginator) {
+            let oldPosition = this.paginator.getCurrentPosition();
+            this.setIFrameSize();
+            this.paginator.goToPosition(oldPosition);
+        } else {
+            this.setIFrameSize();
+        }
+    }
+
+    private handlePreviousChapterClick(event: MouseEvent): void {
+        if (this.previousChapterLink.hasAttribute("href")) {
+            let position = {
+                resource: this.previousChapterLink.href,
+                position: 0
+            }
+            this.navigate(position);
+        }
+        event.preventDefault();
+    }
+
+    private handleNextChapterClick(event: MouseEvent): void {
+        if (this.nextChapterLink.hasAttribute("href")) {
+            let position = {
+                resource: this.nextChapterLink.href,
+                position: 0
+            };
+            this.navigate(position);
+        }
+        event.preventDefault();
+    }
+
+    private handleStartClick(event: MouseEvent): void {
+        if (this.startLink.hasAttribute("href")) {
+            let position = {
+                resource: this.startLink.href,
+                position: 0
+            };
+            this.navigate(position);
+        }
+        event.preventDefault();
+    }
+
+    private handleContentsClick(event: MouseEvent): void {
+        let position = {
+            resource: this.contentsLink.href,
+            position: 0
+        };
+        this.navigate(position);
+        event.preventDefault();
+    }
+
+    private navigate(readingPosition: ReadingPosition): void {
+        this.loading.style.display = "block";
```

Comment:
It would be nice if we displayed this after N milliseconds to avoid it flashing on every load.